botocore/paginate.py

```diff
@@ -623,11 +625,17 @@ def _extract_paging_params(self, kwargs):
             max_items = int(max_items)
         page_size = pagination_config.get('PageSize', None)
         if page_size is not None:
-            if self._pagination_cfg.get('limit_key', None) is None:
+            if self._limit_key is None:
                 raise PaginationError(
                     message="PageSize parameter is not supported for the "
                             "pagination interface for this operation.")
-            page_size = int(page_size)
+            input_members = self._model.input_shape.members
+            limit_key_shape = input_members.get(self._limit_key)
+            if isinstance(limit_key_shape, StringShape):
```

Comment:
Hmm. I'm not sure how I feel about this. Either way this code will have a dependency on `StringShape`. The way you propose will remove the import but the dependency on `StringShape`'s `type_name` being `'string'` is still there. Can you elaborate on why you want to avoid the import?